src/formatter/json_formatter.ts

```diff
@@ -317,20 +318,33 @@ export default class JsonFormatter extends Formatter {
     pickle: IPickle
     gherkinScenarioMap: { [id: string]: IScenario }
   }): IJsonTag[] {
-    return _.map(pickle.tags, (tagData) => {
-      const featureSource = feature.tags.find(
-        (t: ITag) => t.id === tagData.astNodeId
-      )
-      const scenarioSource = gherkinScenarioMap[pickle.astNodeIds[0]].tags.find(
-        (t: ITag) => t.id === tagData.astNodeId
-      )
-      const line = doesHaveValue(featureSource)
-        ? featureSource.location.line
-        : scenarioSource.location.line
-      return {
-        name: tagData.name,
-        line,
-      }
-    })
+    const scenario = gherkinScenarioMap[pickle.astNodeIds[0]]
+
+    const asJsonTag = (tagData: IPickleTag): IJsonTag =>
+      this.getScenarioTag(tagData, feature, scenario)
+
+    return pickle.tags.map(asJsonTag)
+  }
+
+  getScenarioTag(
```

Comment:
Could this method be private?

Reply:
Yes, it could.
Why other methods in that class are not private?

Reply:
Good question!

I guess technically the class could be extended by a user for their own formatter, though I'm not sure why anyone would extend this one.